text/0000-advance-security-disclosure.md

```diff
@@ -0,0 +1,317 @@
+WG: a majority of the WG members needs to be in favor of the organization
+joining, and none of the members needs to be against it. The resulting list
+of organizations receiving advance disclosure will also be public.
+
+In addition to individual organizations, industry-standard mailing lists and
+disclosure venues that are frequented mostly by organizations that would
+otherwise be eligible to be notified can be added to the list, if the WG
+determines the venue is appropriate, and has sufficient security and
+confidentiality practices.
+
+Organizations won't need to pay any fee to receive early notifications, nor
+will they have to be a sponsor to the Rust Foundation. There will be no formal
+NDA in place, only the informal confidentiality pledge.
+
+Once the application is accepted, the organizational contacts will be able to
+request changes to the address receiving the notifications or the list of
+contacts by opening another issue.
+
+### Removal from the list
+
+Organizations can be removed from the list in any of these cases:
+
+* The reason why the organization joined the list is not applicable anymore,
+  for example if they discontinued their distribution. Organizations can
+  re-apply later if they become eligible again.
+
+* The organization did not act on the early notifications it received for an
+  extended period of time. The WG will work with the organization to see if
+  there are valid reasons for not acting on the notifications before removing
+  the organization. Organizations can re-apply later.
+
+* The organization either accidentally violated confidentiality more than once
+  without having taken appropriate steps to prevent it from happening again
+  after the first violation, or intentionally/maliciously violated
+  confidentiality one time. Organizations will only be able to re-apply with
+  full consent from the whole Security Response WG and Rust Core Team.
+
+The decision will be made by the WG, and can be appealed to the Rust Core Team.
+
+### Logistics of the notification
+
+Advance notifications will be sent to all organizations in the list, containing
+the draft advisory and the relevant patches (if available). The timing of the
+notification will depend on the impact of the vulnerability:
+
+* For most vulnerabilities, five business days before public disclosure.
+* For actively exploited critical vulnerabilities, one business day before
+  public disclosure.
+* For actively exploited critical vulnerabilities that do not require a
+  toolchain update to be effectively mitigated, no advance notification will be
+  sent.
+
+Not all vulnerabilities will result in an advance notification being sent.
+Namely, vulnerabilities in the project infrastructure, crates maintained by the
+Rust project, or other projects that are not shipped as part of the toolchain
+will not result in an advance notification, as there would be no update for the
+organizations to prepare in advance.
```

Comment:
I'm not sure this is true? If (say) regex needs a security update, users need to prepare to bump just as much as they'd prepare for a toolchain upgrade, IMO.

I think the guiding line is if the update is something we do or users do: if users need to take any action, there should be a similar process for any of those actions IMO.

Reply:
> I'm not sure this is true? If (say) regex needs a security update, users need to prepare to bump just as much as they'd prepare for a toolchain upgrade, IMO.

This is a relic from a previous draft of this RFC, that didn't include public pre-announcements of vulnerabilities, and doesn't account for [programmerjake](https://github.com/programmerjake)'s mention of distributions also distributing crates in addition to toolchains. I'll update the text shortly.